nsis/src/nsis/file/NsisExecutable.java

```diff
@@ -59,15 +62,17 @@ public static NsisExecutable createNsisExecutable(GenericFactory factory, BytePr
 
 	private void initNsisExecutable(GenericFactory factory, ByteProvider bp, SectionLayout layout)
 			throws IOException, InvalidFormatException {
-		this.reader = new FactoryBundledWithBinaryReader(factory, bp, /* isLittleEndian= */ true);
+		this.reader = new FactoryBundledWithBinaryReader(factory, bp,
+				NsisConstants.IS_LITTLE_ENDIAN);
 		this.headerOffset = findHeaderOffset();
 		initScriptHeader();
-		if ((this.scriptHeader.compressedHeaderSize & FLAG_IS_COMPRESSED) != 0) { // Check if MSB is
-																					// set
-			this.bodyData = decompressData();
-		} else {
-			this.bodyData = this.reader.readNextByteArray(this.scriptHeader.inflatedHeaderSize);
-		}
+		NsisDecompressionProvider decompressionProvider = decompressData(
```

Comment:
See other comment, this can be saved as `this.decompressionProvider`. Then anytime you want the input stream you can call `this.getBlockInputStream()`